text/0000-no-no-std-standard.md

```diff
@@ -0,0 +1,100 @@
+- Feature Name: no_no_std_standard
+- Start Date: 2017-01-03
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Reject crates.io uploads which declare a feature named `no_std`. Instead, suggest changing the crate to use a `std` feature that is enabled by default. This convention is more in line with how dependency resolution works in Cargo, and so enforcing it will lead to more consistent behavior.
+
+Note that the Rust `#![no_std]` attribute is unrelated to this proposal, and will not be affected by it.
+
+# Motivation
+[motivation]: #motivation
+
+Crate features are *additive*. If two crates need different features from the same dependency, then Cargo will take the union of both. In other words, you can only *enable* a feature on a dependency, never disable it.
+
+This can cause issues when one of these features is *negative*. In particular, a crate may provide a `no_std` feature, which restricts its API so that it works with just the `core` library. But because features can only be enabled, not disabled, any consumer which enables `no_std` here will also enable it for every other user of this crate. Most of the time, this is not what the user wants to do.
+
+While there are other examples of negative features out there, `no_std` is by far the most prominent. This is because without knowing about Cargo's algorithms, it would seem logical to toggle `#![no_std]` support using a feature of the same name.
+
+As of this writing, there are [28 crates] on crates.io (0.4% of 7,426 total) with a feature that contains `no_` as a substring. Out of these, 21 of them have `no_std`. As for the remaining seven:
```

Comment:
Are there any other negative-ish feature names or substrings that are commonly used? Ex: `disable`, `off`, `not`, things like that. It feels fragile to not only focus on `no_std` to the exclusion of the other `no_` features, but also ignore any other negative feature names that don't work with the additive property of crate features.... or even a positive name that is used to restrict a crate's API!